functions.opengraph.php

```diff
@@ -100,10 +104,12 @@ function jetpack_og_tags() {
 
 		$tags['og:url']         = get_permalink( $data->ID );
 		if ( ! post_password_required() ) {
-			if ( ! empty( $data->post_excerpt ) ) {
-				$tags['og:description'] = preg_replace( '@https?://[\S]+@', '', strip_shortcodes( wp_kses( $data->post_excerpt, array() ) ) );
+			$excerpt = get_the_excerpt( $data );
+			if ( ! empty( $excerpt ) && ! empty( $data->post_excerpt ) ) {
+				$tags['og:description'] = preg_replace( '@https?://[\S]+@', '', $excerpt
+ );
 			} else {
-				$exploded_content_on_more_tag = explode( '<!--more-->', $data->post_content );
+				$exploded_content_on_more_tag = explode( '<!--more-->', apply_filters( 'the_content', $data->post_content ) );
```

Comment:
Applying `the_content` is problematic since it is the best generic place to stick things at the end of the content (likes/sharing/related posts, etc).

It feels almost like a Core gap -- the need to transform the content versus the need to append the content. Right now, `the_content` does both. For the og use case, we only need the former.

FWIW, Jetpack's Markdown implementation gets around this by, on post save, putting the html into `post_content` and saving the unrendered markdown in `post_content_filtered` so other plugins have the rendered html to transform.

My initial thought is I'm not okay with re-adding the `the_content` apply_filter since it would re-introduced the previous problem but would be open to entertaining a filter that a plugin could hook onto specifically for `jetpack_og_description_content` (or something similar).

Thoughts?